sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/SqlDelightReferenceContributor.kt

```diff
@@ -74,3 +74,17 @@ internal class SqlDelightReferenceContributor : PsiReferenceContributor() {
     private fun typeForThisPackage(file: SqlDelightFile) = "${file.packageName}.${element.text}"
   }
 }
+
+private fun getKotlinTopLevelTypeAliasFqNameIndex(): KotlinTopLevelTypeAliasFqNameIndex {
+  // read the INSTANCE variable reflectively first (newer Kotlin plugins)
+  try {
+    val instanceField = KotlinTopLevelTypeAliasFqNameIndex::class.java.getField("INSTANCE")
+    val instance = instanceField.get(null)
+    if (instance is KotlinTopLevelTypeAliasFqNameIndex) {
```

Comment:
Should not we throw an exception in the else block in case `instance` is of any type other than `KotlinTopLevelTypeAliasFqNameIndex`? So if in the future it breaks again, we at least get a proper stacktrace

Reply:
hmm, I think that case overall is highly unlikely: `<T>.INSTANCE` field exists but is not of type `T`.

To make it safer we could reflectively check for `Companion` + `getInstance` method though I guess and then throw if neither `INSTANCE` field or `$Companion.getInstance` is present?

Reply:
Fair enough!